packages/matrix-client-server/src/account/whoami.ts

```diff
@@ -13,17 +13,6 @@ const whoami = (clientServer: MatrixClientServer): expressAppHandler => {
       clientServer.matrixDb
         .get('users', ['name', 'is_guest'], { name: data.sub })
         .then((rows) => {
-          if (rows.length === 0) {
-            // istanbul ignore next // TODO : Test this after implementing /register endpoint
-            send(
-              res,
-              403,
-              errMsg(
-                'forbidden',
-                'The appservice cannot masquerade as the user or has not registered them.'
-              )
-            )
-          }
```

Comment:
Yes since the users table in the matrixDB is filled at the same time as the user_ips table by the /register endpoint, and authenticate checks that the access_token corresponding to that userId is in the user_ips table, so if the request is correctly authenticated, it means that the user_ips table was filled which necessarily means the users table was also filled with the same name since the only endpoint that fills them is /register (for now, if another endpoint interacts with one of those two tables in the future, I will change this).